From 96fa0fba20173a7677b0a93c7edbc1287805d727 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Fri, 23 Oct 2026 00:38:43 +0200
Subject: [PATCH 1/1] Remove Software center filter option

Remove the Sources filter with the Software center option and the
suggestions of the applications available in the software center, which
is not available. Without the Sources filter, the local applications
are always active.

Signed-off-by: c4pp4
---
 src/daemon.vala | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/daemon.vala b/src/daemon.vala
index 5630f2b..1f632df 100644
--- a/src/daemon.vala
+++ b/src/daemon.vala
@@ -264,6 +264,7 @@ namespace Unity.ApplicationsLens {
       }
 
       /* Sources */
+      /*
       {
         var filter = new CheckOptionFilter ("unity-sources", _("Sources"));
         filter.sort_type = Unity.OptionsFilter.SortType.DISPLAY_NAME;
@@ -280,6 +281,7 @@ namespace Unity.ApplicationsLens {
 
         filters.add (filter);
       }
+      */
 
       return filters;
     }
@@ -843,15 +845,18 @@ namespace Unity.ApplicationsLens {
 
     private bool local_apps_active ()
     {
+      /*
       var filter = search_context.filter_state.get_filter_by_id ("unity-sources") as Unity.OptionsFilter;
       if (filter.filtering)
       {
         var option = filter.get_option ("local");
         return option == null || option.active;
       }
+      */
       return true;
     }
 
+    /*
     private bool usc_apps_active ()
     {
       var filter = search_context.filter_state.get_filter_by_id ("unity-sources") as Unity.OptionsFilter;
@@ -862,6 +867,7 @@ namespace Unity.ApplicationsLens {
       }
       return true;
     }
+    */
 
     /* Returns TRUE if application is NOT installed */
     public bool filter_cb (Unity.Package.PackageInfo pkginfo)
@@ -962,6 +968,7 @@ namespace Unity.ApplicationsLens {
       scope.purchase_info = new PurchaseInfoHelper ();
 
       /* If we don't have a search we display 6 random apps */
+      /*
       if (usc_apps_active () && scope.display_available_apps && scope.pkgsearcher != null)
       {
         if (has_search)
@@ -977,7 +984,7 @@ namespace Unity.ApplicationsLens {
           debug ("Entry search listed %i Available apps in %fms for query: %s",
                  pkgresults.num_hits, timer.elapsed ()*1000, pkg_search_string);
         }
-        else if (has_filter) /* Empty search string + active filters should get lots of results from selected categories */
+        else if (has_filter) // Empty search string + active filters should get lots of results from selected categories
         {
           timer.start ();
           string? filter_query = XapianUtils.prepare_pkg_search_string (search_string, type_filter);
@@ -1023,6 +1030,7 @@ namespace Unity.ApplicationsLens {
                  hits, timer.elapsed ()*1000);
         }
       }
+      */
 
       /* XXX:
       if (model.get_n_rows () == 0)
-- 
2.55.0

